### PR TITLE
Baseline against recent API changes

### PR DESCRIPTION
**Pull request by tommyb82**
As discussed, this includes a number of changes to bring the codebase up to date with the latest API changes, plus:

- Reconfigured Agreement Service client to use `WebClient` for consistency with the Jaggaer integration (and renamed a couple of classes)
- Beginnings of SCC-439 response mapping (TODO)
- Rejigged some of the code in `ProcurementEventService` to hopefully make things a little easier to understand and maintain
- `ID` -> `Id` property name suffix updates
- JPA fix in `ProcurementEvent`
- Introduces a `ValidationService` component

The basics for SCC-439 are in there too - but needs fleshing out now the API is stable.  Be easier to manage that once this lot is settled.

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/entity/ProcurementProject.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
   /**
    * Builds an instance from basic details
    *
-   * @param agreementDetails
+   * @param projectRequest
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/entity/ProcurementProject.java:59]: should this still be `agreementDetails`?..

Comment (tommyb82) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/entity/ProcurementProject.java:59]: Aye - probably 👍

**Review by adrianmilne — APPROVED** [review]
👍 Excellent stuff - great job tidying that up - ran locally and created a project/event 🥳 

(Guess we just need a corresponding update in the Postman Collection to change `ID` to `Id`)

**Comment by tommyb82** [comment]
> 👍 Excellent stuff - great job tidying that up - ran locally and created a project/event 🥳
> 
> (Guess we just need a corresponding update in the Postman Collection to change `ID` to `Id`)

Yep - I'll get an updated Postman collection committed today 👍

**Pull request merged**